datafusion/core/tests/parquet/arrow_statistics.rs

```diff
@@ -1984,7 +1981,96 @@ async fn test_struct() {
     }
     .run();
 }
+// test nested struct
+#[tokio::test]
+async fn test_nested_struct() {
+    // This creates a parquet file with 1 column named "nested_struct"
+    // The file is created by 1 record batch with 3 rows in the nested struct array
+    let reader = TestReader {
+        scenario: Scenario::StructArrayNested,
+        row_per_group: 5,
+    }
+    .build()
+    .await;
 
+    // Expected minimum and maximum values for nested struct fields
+    let inner_min = StructArray::from(vec![
+        (
+            Arc::new(Field::new("b", DataType::Boolean, false)),
+            Arc::new(BooleanArray::from(vec![Some(false)])) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("c", DataType::Int32, false)),
+            Arc::new(Int32Array::from(vec![Some(42)])) as ArrayRef,
+        ),
+    ]);
+    let inner_max = StructArray::from(vec![
+        (
+            Arc::new(Field::new("b", DataType::Boolean, false)),
+            Arc::new(BooleanArray::from(vec![Some(true)])) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("c", DataType::Int32, false)),
+            Arc::new(Int32Array::from(vec![Some(44)])) as ArrayRef,
+        ),
+    ]);
+
+    let inner_fields = Fields::from(vec![
+        Field::new("b", DataType::Boolean, false),
+        Field::new("c", DataType::Int32, false),
+    ]);
+
+    // Expected minimum outer struct
+    let expected_min_outer = StructArray::from(vec![
+        (
+            Arc::new(Field::new(
+                "inner_struct",
+                DataType::Struct(inner_fields.clone()),
+                false,
+            )),
+            Arc::new(inner_min) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("outer_float", DataType::Float64, false)),
+            Arc::new(Float64Array::from(vec![Some(5.0)])) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("outer_boolean", DataType::Boolean, false)),
+            Arc::new(BooleanArray::from(vec![Some(false)])) as ArrayRef,
+        ),
+    ]);
+
+    // Expected maximum outer struct
+    let expected_max_outer = StructArray::from(vec![
+        (
+            Arc::new(Field::new(
+                "inner_struct",
+                DataType::Struct(inner_fields),
+                false,
+            )),
+            Arc::new(inner_max) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("outer_float", DataType::Float64, false)),
+            Arc::new(Float64Array::from(vec![Some(7.0)])) as ArrayRef,
+        ),
+        (
+            Arc::new(Field::new("outer_boolean", DataType::Boolean, false)),
+            Arc::new(BooleanArray::from(vec![Some(true)])) as ArrayRef,
+        ),
+    ]);
+
```

Comment:
I wondered if we needed a struct array with top-level nulls. But for the stats, it shouldn't matter whether the null is coming from the top level or the nested arrays. That would be testing the how the stats are calculated. (This assumes I'm interpreting the way nulls are handled correctly.)

But it might be worth having some null counts to see that they get calculated correctly?

Reply:
That's what I concern about. Since it is hard and seems meaningless to have a top-level nullcount. The reason I leave it like this is just to follow the test pattern.

Reply:
This blog post provides good examples on when you'd represent nulls at different levels: https://arrow.apache.org/blog/2022/10/08/arrow-parquet-encoding-part-2/

Reply:
I was going to read the valid mask but it actually contains in the data page, not sure whether we need to read the actual data at this time.

Reply:
I don't _think_ you need to do anything differently. I was trying to find some tests in the arrow crate for when the statistics are written but I haven't been able to find any for writing nested structs.

Reply:
Here's some tests in the arrow crate that are writing a 2 level struct: https://github.com/apache/arrow-rs/blob/master/parquet/src/arrow/arrow_writer/mod.rs#L1516-L1618 Not sure if that's helpful or not.

Reply:
Sure, I'll take a look at it.